invenio_drafts_resources/records/systemfields/versions.py

```diff
@@ -166,6 +163,19 @@ def __repr__(self):
         )
 
 
+class VersionsFieldContext(SystemFieldContext):
+    """Version field context."""
+
+    @property
+    def model_cls(self):
+        """Versions model class."""
+        return self.record_cls.versions_model_cls
+
+    def resolve(self, *, parent_id=None):
+        """Resolve a versions state from a parent ID."""
+        return self.model_cls.query.filter_by(parent_id=parent_id).one()
```

Comment:
should we handle the possible `None` value of the `parent_id` arg? Or will it raise a `NoResultFound` is the value is `None`?

Reply:
Actually that was a leftover, originally I went a bit ambitious and wanted to also allow passing any version's `record_id` like:

```python
    def resolve(self, *, parent_id=None, record_id=None):
        """Resolve a versions state from a parent ID."""
        assert parnet_id or record_id
        if parent_id:
            return self.model_cls.query.filter_by(parent_id=parent_id).one()
        # ... try to resolve using `record_id`
        ...
```

But there was no real use-case for that in the codebase, so I just left it out. 

I can remove the `=None` default, from the args. Note also that I used the `*` in the definition, so that you always have to pass it as a keyword argument:

```python
# This won't work:
RDMRecord.version.resolve("<some_id>")

# You need to be explicit like so:
RDMRecord.version.resolve(parent_id="<some_id>")
```